presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/TestingArrowServer.java

```
@@ -0,0 +1,308 @@
+            FlightEndpoint endpoint = new FlightEndpoint(new Ticket(flightDescriptor.getCommand()));
+            return new FlightInfo(schema, flightDescriptor, Collections.singletonList(endpoint), -1, -1);
+        }
+        catch (Exception e) {
+            logger.error(e);
+            throw new RuntimeException("Failed to retrieve FlightInfo", e);
+        }
+    }
+
+    @Override
+    public Runnable acceptPut(CallContext callContext, FlightStream flightStream, StreamListener<PutResult> streamListener)
+    {
+        return null;
+    }
+
+    @Override
+    public void doAction(CallContext callContext, Action action, StreamListener<Result> streamListener)
+    {
+        try {
+            String jsonRequest = new String(action.getBody(), StandardCharsets.UTF_8);
+            JsonNode rootNode = objectMapper.readTree(jsonRequest);
+            String schemaName = rootNode.get("interactionProperties").get("schema_name").asText(null);
+
+            String query;
+            if (schemaName == null) {
+                query = "SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA";
+            }
+            else {
+                schemaName = schemaName.toUpperCase();
+                query = "SELECT TABLE_NAME FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA='" + schemaName + "'";
+            }
+            ResultSet rs = connection.createStatement().executeQuery(query);
+            List<String> names = new ArrayList<>();
+            while (rs.next()) {
+                names.add(rs.getString(1));
+            }
+
+            String jsonResponse = objectMapper.writeValueAsString(names);
+            streamListener.onNext(new Result(jsonResponse.getBytes(StandardCharsets.UTF_8)));
+            streamListener.onCompleted();
+        }
+        catch (Exception e) {
+            streamListener.onError(e);
+        }
+    }
+
+    @Override
+    public void listActions(CallContext callContext, StreamListener<ActionType> streamListener)
```

Please throw UnsupportedOperationException

---

Updated